### DISCUSSION TITLE
Can Codespaces support Vnet?

### DISCUSSION THREAD

**Discussion by luxu-ms** [community/community#10477] [Codespaces]
During the conversation with some customers, they are happy to develop in the container of Codespaces. But according to their policy, they need to access to their own git repository/Nexus/Harbor. That means they only use the Codespaces as the host for the containers, but they still need to access back to their own enterprise network.

Question: Is there any chance that Codespaces will support Vnet? Thus let the customer connect back to their own enterprise/private network?

**Comment by tanmayeekamath** [comment]
Hey @luxu-ms! Thanks for reaching out! 
As of today, here are some of the options to access on-prem resources from a Codespace: https://docs.github.com/en/codespaces/developing-in-codespaces/connecting-to-a-private-network

That being said, I would love to better understand your needs for this scenario. Are your resources behind a VPN on prem or do you also have a need to access cloud based resources? Do you have existing VNETs that you would expect to work with a Codespaces? 

We're also exploring a solution that would involve the Codespace using the developer's local machine as a means to access any on-prem resources that the machine already has access to. Would this work as a potential solution to address your need?

**Reply by luxu-ms** [reply]
Hi Kamath,

Thanks for your response.
The typical scenario is that the customer is very interested in using the codespaces. But due to their policy, they need to hold their private code repository (Bitbucket, Gitlab) in their on-premises network. So the codespaces need to connect back to their on-premises network. Instead of VPN, they hope there is the VNet can be provided that codespaces can directly access to their on-premises resources.

**Reply by hfhbd** [reply]
According to the roadmap, there was a private beta and it is marked as shipped: https://github.com/github/roadmap/issues/534
Is there any update when this feature will be available as public preview? Or how can you join the private beta? We would love to use Codespaces in our company, but we need access to our internal network, like we do use Azure VNet to connect the GitHub runners.

**Comment by Jamonygr** [comment]
This is still a gap for teams that want Codespaces to behave like a dev VM inside their enterprise network. The current private-network options are more like gateway or VPN-style access from the developer side; they are not the same as attaching a Codespace directly to your Azure VNet. For private Git, Nexus, Harbor, or Azure private endpoints, the practical workaround is usually a corporate dev VM/VDI, a self-hosted runner-like environment, or a brokered tunnel that the security team owns.
